.github/workflows/beam_PreCommit_SQL_Java11.yml

```diff
@@ -0,0 +1,96 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SQL Java11 PreCommit 
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: ['sdks/java/extensions/sql/**','.github/workflows/beam_PreCommit_SQL_Java11.yml']
+  pull_request_target:
+    branches: ['master', 'release-*']
+    paths: [sdks/java/extensions/sql/**]
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '* */6 * * *'
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+    actions: write
+    pull-requests: read
+    checks: read
+    contents: read
+    deployments: read
+    id-token: none
+    issues: read
+    discussions: read
+    packages: read
+    pages: read
+    repository-projects: read
+    security-events: read
+    statuses: read
+
+jobs:
+  beam_PreCommit_SQL_Java11:
+    if: github.event_name == 'push' || github.event_name == 'pull_request' || github.event_name == 'schedule' || github.event.comment.body == 'Run SQL_Java11 PreCommit'
```

Comment:
We need to use the `pull_request_target` event

Reply:
> Сould you list the conditions on each line to make it more readable? Here is an example - https://github.com/apache/beam/blob/master/.github/workflows/beam_PreCommit_Java_Examples_Dataflow.yml#L65-L69

done